main.go

```diff
@@ -296,9 +316,82 @@ func (s XcodeArchiveStep) ProcessInputs() (Config, error) {
 		config.ArtifactName = productName
 	}
 
+	codesignManager, err := s.createCodesignManager(config)
```

Comment:
why is it done in the ProcessInputs? previously this function only converted Step Inputs to a Config, i think it should go to XcodeArchiveStep.Run

Reply:
CodesignManager is created here because the creation of this instance involves a lot of input parsing and validation. It's only the instance creation that happens in `ProcessInputs()`, `codesignManager.PrepareCodesigning()` (which is the heavy operation) happens in `Run()`